### ISSUE TITLE
fix: stack overflow in GPU batch processor (probador c=1 crash)

### ISSUE THREAD

**Issue by noahgift** [paiml/realizar#171]
## Bug

`apr serve run --gpu --batch` crashes with stack overflow on first request:

```
thread 'tokio-rt-worker' (2345919) has overflowed its stack
fatal runtime error: stack overflow, aborting
```

## Reproduction

```bash
apr serve run model.gguf --gpu --batch --port 8098
curl http://127.0.0.1:8098/v1/chat/completions \
  -d '{"model":"q","messages":[{"role":"user","content":"Hi"}],"max_tokens":5,"temperature":0}'
```

Server crashes immediately on first request.

## Root cause (five-whys)

1. Why crash? → Stack overflow in tokio-rt-worker
2. Why overflow? → Deep recursion or large stack allocation in batch decode path
3. Why large? → Logits buffer (vocab_size=151936 × 4 bytes = 607KB) likely on stack
4. Why stack? → Local `Vec<f32>` or array allocation inside async context
5. Root cause: **large stack frame in batch decode exceeds tokio default worker stack (8MB)**

## Fix

Either:
- Box-allocate large buffers (logits, hidden states) in batch path
- Increase tokio worker stack size: `tokio::runtime::Builder::new_multi_thread().thread_stack_size(16 * 1024 * 1024)`

## Impact

Blocks F-SCALE-01 re-test with batch mode (candle-vs-apr).
probador llm load --concurrency 4 with --batch gets 100% failure rate.

Refs candle-vs-apr F-SCALE-01, PMAT-037
